### PR TITLE
Linux tests

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

File: coding/zlib.hpp [hunk 1]
```diff
@@ -171,7 +171,7 @@ class ZLib
     }
 
     processor.MoveOut(out);
-    return true;
+    return processor.ConsumedAll();
```
Comment (mpimenov) [inline coding/zlib.hpp]: Add a test for new behaviour, please.

**Comment by tatiana-yan** [comment]
http://jenkins-master.mapsme.cloud.devmail.ru/jenkins/job/LinuxTests/79/

File: coding/zlib.hpp (hunk 1 above)
Comment (tatiana-yan) [reply coding/zlib.hpp]: добавила

**Comment by tatiana-yan** [comment]
JTMAC

**Comment by tatiana-yan** [comment]
http://jenkins-master.mapsme.cloud.devmail.ru/jenkins/job/LinuxTests/80/

File: platform/http_client_curl.cpp [hunk 2]
```diff
@@ -285,13 +285,16 @@ bool HttpClient::RunHttpRequest()
...
-  auto const it = m_headers.find("content-encoding");
-  if (it != m_headers.end())
+  for (auto const & header : headers)
   {
-    m_serverResponse = Decompress(m_serverResponse, it->second);
-    LOG(LDEBUG, ("Response with", it->second, "is decompressed."));
+    if (strings::EqualNoCase(header.first, "content-encoding") &&
+        !strings::EqualNoCase(header.second, "identity"))
+    {
+      m_serverResponse = Decompress(m_serverResponse, header.second);
```
Comment (mpimenov) [inline platform/http_client_curl.cpp]: Please add a todo and an UNUSED_VALUE to Line 161 in this file (in Decompress).

Comment (tatiana-yan) [reply platform/http_client_curl.cpp]: добавила

**Pull request merged**